cms/djangoapps/contentstore/views/transcript_settings.py

```diff
@@ -254,3 +261,32 @@ def transcript_upload_handler(request, course_key_string):
             )
 
     return response
+
+
+@login_required
+@require_http_methods(["DELETE"])
+def transcript_delete_handler(request, course_key_string, edx_video_id, language_code):
+    """
+    View to delete a transcript file.
+
+    Arguments:
+        request: A WSGI request object
+        course_key_string: Course key identifying a course.
+        edx_video_id: edX video identifier whose transcript need to be deleted.
+        language_code: transcript's language code.
+
+    Returns
+        - A 404 if the corresponding feature flag is disabled
+        - A 200 if transcript is deleted without any error(s)
+    """
+    # Check whether the feature is available for this course.
+    course_key = CourseKey.from_string(course_key_string)
+    video_transcripts_enabled = VideoTranscriptEnabledFlag.feature_enabled(course_key)
+    # User needs to be either staff or a course's staff.
+    is_staff = request.user.is_staff or CourseStaffRole(course_key).has_user(request.user)
```

Comment:
what about `CourseInstructorRole`? Should they be also able to use this endpoint?

Reply:
I've changed it.